var/spack/repos/builtin/packages/intel-mpi-benchmarks/package.py

```diff
@@ -18,74 +18,103 @@ class IntelMpiBenchmarks(MakefilePackage):
     """
 
     homepage = "https://software.intel.com/en-us/articles/intel-mpi-benchmarks"
-    url = "https://github.com/intel/mpi-benchmarks/archive/IMB-v2019.5.tar.gz"
-
+    url = "https://github.com/intel/mpi-benchmarks/archive/IMB-v2021.3.tar.gz"
     maintainers = ["carsonwoods"]
 
+    version("2021.3", sha256="9b58a4a7eef7c0c877513152340948402fd87cb06270d2d81308dc2ef740f4c7")
+    version("2021.2", sha256="ade3bfe18b4313a31fc09f0bf038e0a6c169c4145089bfc6f1f827687b81be6a")
+    version("2021.1", sha256="9089bb81e3091af3481e03b898b339fb2d9fb6574d4ef059adb1f5410112b23a")
     version("2019.6", sha256="1cd0bab9e947228fced4666d907f77c51336291533919896a923cff5fcad62e9")
     version("2019.5", sha256="61f8e872a3c3076af53007a68e4da3a8d66be2ba7a051dc21e626a4e2d26e651")
     version("2019.4", sha256="aeb336be10275c1a2f579b491b6631122876b461ac7148b1d0764f13b7552690")
     version("2019.3", sha256="4f256d11bfed9ca6166548486d61a062e67be61f13dd9f30690232720e185f31")
     version("2019.2", sha256="0bc2224a913073aaa5958f6ae08341e5fcd39cedc6722a09bfd4a3d7591a340b")
+    version("2019.1", sha256="fe0d065b9936b6943ea83cb3d00aede43b17565285c6b1791fee8e340853ef79")
+    version("2019.0", sha256="1c7d44aa7fd86ca84ac7cae1a69a8426243048d6294582337f1de7b4ffe68d37")
+    version("2018.1", sha256="718a4eb155f18cf15a736f6496332407b5837cf1f19831723d4cfe5266c43507")
+    version("2018.0", sha256="2e60a9894a686a95791be2227bc569bf81ca3875421b5307df7d83f885b1de88")
+
+    depends_on("mpi", when="@2019:")
+    depends_on("intel-mpi", when="@2018")
+    depends_on("gmake", type="build", when="@2018")
 
-    depends_on("mpi")
+    conflicts(
+        "^openmpi",
+        when="@:2019.0",
+        msg="intel-mpi-benchmarks <= v2019.0 cannot be built with OpenMPI, "
+        "please specify a different MPI implementation",
+    )
```

Comment:
So if I'm understanding correctly, 2018 works with intel-mpi, 2019.0 works with all MPI except openmpi, and 2019.1+ works with all MPI? (obviously you haven't tested all possible MPI providers, but you know what I mean)

Reply:
Yes that seems to be the case. I suspect that some more elaborate modifications to the build process could enable other MPI implementations on older versions, however documentation for the package is unclear so I'm sticking with what works out of the box.